app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,77 @@
+class TweetsController < ApplicationController
+    use Rack::Flash
+    get '/tweets' do
+        if logged_in?
+            @user = User.find_by_id(session[:user_id])
+            erb :'/tweets/tweets'
+        else
+            redirect to "/login"
+        end
+    end
+
+    get '/tweets/new' do
+        if logged_in?
+            erb :'/tweets/create_tweet'
+        else
+            redirect "/login"
+        end
+    end
+
+    post '/tweets' do
+        if !params[:content].empty?
+            @tweet = Tweet.create(:content => params[:content])
+            @tweet.user = current_user
+            @tweet.save
+            @user = @tweet.user
+            erb :"/tweets/tweets"
+        else
+            redirect "/tweets/new"
+        end
+    end
+
+    get '/tweets/:id' do
+        if logged_in?
+            @tweet = Tweet.find_by_id(params[:id])
+            erb :'/tweets/show_tweet'
+        else
+            redirect "/login"
+        end
+    end
+
+    get '/tweets/:id/edit' do
+        if logged_in?
+            @tweet = Tweet.find_by_id(params[:id])
+            if @tweet.user.username == current_user.username
+                erb :'/tweets/edit_tweet'
+            else
+                flash[:message] = "You are not #{@tweet.user.username}!"
+                erb :'/tweets/tweets'
+            end
+        else
+            redirect "/login"
+        end
+    end
+
+    patch '/tweets/:id' do
+        @tweet = Tweet.find_by_id(params[:id])
+        if !params[:content].empty?
+            @tweet.update(:content => params[:content])
+            @user = current_user
+            erb :"/tweets/tweets"
+        else
+            redirect "/tweets/#{@tweet.id}/edit"
+        end
+    end
+
+    get '/tweets/:id/delete' do
+        @tweet = Tweet.find_by_id(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.